kview/src/main/java/dev/ikm/komet/kview/controls/skin/PublicIDControlSkin.java

```diff
@@ -107,8 +115,19 @@ public PublicIDControlSkin(PublicIDControl control) {
         // subscribe to changes to the publicIdProperty in the PublicIDControl
         subscription = control.publicIdProperty().subscribe(publicId -> {
             identifier = publicId;
-            publicIdTextField.setText(publicId);
-            publicIdTooltip.setText(publicId);
+            publicIdTooltip.setText(identifier);
+            publicIdTextField.setText(identifier);
+
+            Platform.runLater(() -> {
+                // set the preferredWidth of the TextField to completely show the identifier text
+
+                Text text = new Text(identifier); // Create a temporary Text node with the new text
+                text.setFont(publicIdTextField.getFont()); // Set the same font as the TextField
+                double width = text.getLayoutBounds().getWidth() +
+                        publicIdTextField.getPadding().getLeft() +
+                        publicIdTextField.getPadding().getRight() + 2d; // Add padding and a small buffer
+                publicIdTextField.setPrefWidth(width);
```

Comment:
Imo, We should handle this another way. 
publicIdTextField should have css applied and examine the outer parent. I can't remember but I think it is a gridpane cell. could be a colspan.